internal/vxproto/module.go

```diff
@@ -420,23 +434,39 @@ func (ms *moduleSocket) sendPacket(ctx context.Context, dst string, pType Packet
 		Payload: payload,
 	}
 	if ms.IProtoIO == nil {
-		return fmt.Errorf("the ProtoIO is nil")
+		return ErrProtoIOUnset
 	}
-	if ms.HasIMCTokenFormat(dst) {
-		idstms := ms.GetIMCModule(dst)
+	sendViaIMC := func(token string) error {
+		idstms := ms.GetIMCModule(token)
 		if idstms == nil {
-			return fmt.Errorf("destination is unreachable")
+			return ErrDstUnreachable
 		}
 		dstms, ok := idstms.(*moduleSocket)
 		if !ok {
-			return fmt.Errorf("destination is not a module socket")
+			return ErrDstMalformed
 		}
 		packet.Module = dstms.GetName()
 		packet.Src = ms.GetIMCToken()
 		packet.ackChan = make(chan struct{}, 1)
 		packet.ctx = ctx
 		return dstms.recvPacket(ctx, packet)
 	}
+	if ms.HasIMCTopicFormat(dst) {
+		iti := ms.GetIMCTopic(dst)
+		if iti == nil {
+			return ErrTopicUnreachable
+		}
+		var lastErr error
+		for _, token := range iti.GetSubscriptions() {
+			if err := sendViaIMC(token); err != nil {
```

Comment:
In Go 1.20 (Feb '23) it would be possible to join multiple errors - https://tip.golang.org/doc/go1.20#errors - Probably it would be nice to leave a comment for future change.

Reply:
I rewrote this part last commit, now there is joining errors one by one